public/components/metrics/sidebar/metric_name.tsx

```diff
@@ -11,10 +11,15 @@ import { OBSERVABILITY_CUSTOM_METRIC } from '../../../../common/constants/metric
 
 const MetricIcon = ({ metric }) => {
   const metricIcons = useSelector(metricIconsSelector);
-  const iconMeta = metricIcons[metric.catalog];
-  if (metric.catalog === OBSERVABILITY_CUSTOM_METRIC)
+  const iconMeta = metricIcons[metric?.catalog];
```

Comment:
This logic needs to be moved closer to the source of icons.. metricSlice.
At least wrap this in a function, so it can tell it's own story (use creative function naming)
Right now these operations and the if conditions don't make any sense, even in this context.